modules/core/include/opencv2/core/utility.hpp

```diff
@@ -714,9 +714,27 @@ void Mat::forEach_impl(const Functor& operation) {
 /////////////////////////// Synchronization Primitives ///////////////////////////////
 
 #if !defined(_M_CEE)
+#ifndef OPENCV_DISABLE_THREAD_SUPPORT
```

Comment:
@alalek  - I was wondering, should I turn these in `#if !OPENCV_DISABLE_THREAD_SUPPORT` and set the macro to 0 by default, like for the filesystem on?

Francesco

Reply:
In general in makes sense (at least to catch typos through `-Wundef`)
However, we don't use 0/1 "mandatory" values on macros with "_DISABLE_" prefix.
Filesystem's macro name is `OPENCV_HAVE_FILESYSTEM_SUPPORT`.
If you want to switch then it makes sense to rename the macro too (C++ only part without CMake names)